rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/rascal/RascalTextDocumentService.java

```diff
@@ -285,6 +296,56 @@ public CompletableFuture<Either<List<? extends Location>, List<? extends Locatio
             ;
     }
 
+    private ITree getCursorFromPosition(ISourceLocation file, ITree moduleTree, Position p) {
```

Comment:
maybe this function should get a better name? something like `findQualifiedNameUnderCursor`?